pkg/certs/certgen.go

```diff
@@ -0,0 +1,272 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package certs
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha1" // nolint:gosec
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"time"
+)
+
+const (
+	// DefaultContourServiceName holds the default service name
+	// used for the Contour Kubernetes service. This value is added
+	// to the Contour certificate's Subject Alt Names.
+	DefaultContourServiceName = "contour"
+
+	// DefaultEnvoyServiceName holds the default service name
+	// used for the Envoy Kubernetes service. This value is added
+	// to the Envoy certificate's Subject Alt Names.
+	DefaultEnvoyServiceName = "envoy"
+
+	// DefaultCertificateLifetime holds the default certificate lifetime
+	// (in days).
+	DefaultCertificateLifetime = 365
+
+	// DefaultNamespace where Contour is deployed. This value is added
+	// to the certificates Subject Alt Names.
+	DefaultNamespace = "projectcontour"
+
+	// DefaultDNSName holds the Kubernetes local dns suffix name
+	// specific to the cluster where Contour is deployed and is used when
+	// configuring Subject Alt Names on the certificates.
+	DefaultDNSName = "cluster.local"
```

Comment:
Worth adding a flag to make this configurable? Or should that maybe go in a separate issue/PR

Reply:
I think that's a good separate issue. I was trying to make this super generic since it's public and harder to change after committed.